conv_model_bwd.cpp

```diff
@@ -415,6 +415,10 @@ int conv_benchmark(int argc, char** argv) {
     } else {
       tensor_copy_NCHWc_to_NCHW ((float*)input_libxsmm, naive_input_opt, N, C, ifhp, ifwp, bc);
     }
+    /* If non 1x1 and multiple h in gemm, then make sure that we zero out the rims... */
+    if ((R != 1 || S != 1) && (h_in_gemm > 1)) {
```

Comment:
This is what was missing after h_in_gemm support has been added to conv bwd.

Reply:
I am not sure this is the right fix... this may make the correctness test pass, but the actual optimized code should out the rims  like here:
https://github.com/libxsmm/parlooper/blob/8e9bf08d826e7dbf2cbcc86e0349b5bb179d8941/conv_model_bwd.cpp#L362

Reply:
Don't we say that convolution is allowed to produce anything in the padded pixels? We definitely do for forward so I thought same applies to backward by data.

Reply:
OK, the idea we discussed in person indeed is that later parts of the code might be aware of this polluted rims and ignore them (or zero them if anyways are going to touch them). However, to make the code meaningful in a standalone fashion we should by default zero the rims in this case (do so in fwd pass too); and we can add a flag "avoid_zeroing_polluted_rim" that allows to bypass the zeroing in the optimized code (and does the zeroing  in the copy-out just to pass correctness. Does it make sense?

Reply:
Yes, I had the exact same thought when I saw you previous comment. Let me update the PR with this extra flag.

Reply:
Updated the PR, please check if it matches your expectations.

Reply:
Looks good to me, thanks!